versioned/storage/jdbc/src/main/java/org/projectnessie/versioned/storage/jdbc/AbstractJdbcPersist.java

```diff
@@ -508,6 +508,13 @@ private boolean[] upsertObjs(
               if (updated[i] == 1) {
                 r[batchIndexToObjIndex.get(i)] = true;
               }
+              if (updated[i] == PreparedStatement.SUCCESS_NO_INFO) {
```

Comment:
This is a consequence of how MariaDB sometimes reports a successful batch: it returns `SUCCESS_NO_INFO` by default, unless `useBulkStmtsForInserts` is set to false. I think it's wrong to ignore this here. For now I'm throwing ISE, but if #8533 is merged we could instead throw `UnknownOperationResultException`.

Reply:
This snippet has been moved to #8551.